db/python/layers/seqr.py

```diff
@@ -392,71 +411,75 @@ async def update_es_index(
         fn_path = os.path.join(SEQR_MAP_LOCATION, filename)
         # pylint: disable=no-member
 
+        with AnyPath(fn_path).open('w+') as f:  # type: ignore
+            f.write('\n'.join(rows_to_write))
+
         alayer = AnalysisLayer(connection=self.connection)
-        es_index_analyses = await alayer.query(
-            AnalysisFilter(
-                project=GenericFilter(eq=self.connection.project),
-                type=GenericFilter(eq='es-index'),
-                status=GenericFilter(eq=AnalysisStatus.COMPLETED),
-                meta={'sequencing_type': GenericFilter(eq=sequencing_type)},
+        for es_index_type in es_index_types:
+            es_index_analyses = await alayer.query(
+                AnalysisFilter(
+                    project=GenericFilter(eq=self.connection.project),
+                    type=GenericFilter(eq='es-index'),
+                    status=GenericFilter(eq=AnalysisStatus.COMPLETED),
+                    meta={
+                        'sequencing_type': GenericFilter(eq=sequencing_type),
+                        'stage': GenericFilter(eq=ES_INDEX_STAGES[ES_INDEX_DATASET_TYPES[es_index_type]])
+                    },
+                )
             )
-        )
 
-        es_index_analyses = sorted(
-            es_index_analyses,
-            key=lambda el: el.timestamp_completed,
-        )
+            es_index_analyses = sorted(
+                es_index_analyses,
+                key=lambda el: el.timestamp_completed,
+            )
 
-        if len(es_index_analyses) == 0:
-            return ['No ES index to synchronise']
+            if len(es_index_analyses) == 0:
+                return ['No ES index to synchronise']
```

Comment:
I think you'll want to factor this out RE my comment above, but this would cause a hard return if an earlier es-index type doesn't have an es-index to sync, rather than skip and report back.